trie/stacktrie.go

```diff
@@ -363,6 +361,15 @@ func (st *StackTrie) hash() []byte {
 func (st *StackTrie) Hash() (h common.Hash) {
 	st.val = st.hash()
 	st.nodeType = hashedNode
+	if len(st.val) < 32 {
+		ret := make([]byte, 32)
+		h := newHasher(false)
+		defer returnHasherToPool(h)
+		h.sha.Reset()
+		h.sha.Write(h.tmp)
+		h.sha.Read(ret)
+		return common.BytesToHash(ret)
```

Comment:
shouldn't you forcibly set the `st.val` to the forced hash aswell? 
also, does this somehow screw up Commit (hashing a preimage but not sending it out for Commit)